engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -167,13 +168,20 @@ class WorkflowManagerActor(params: WorkflowManagerActorParams)
     /*
      Commands from clients
      */
-    case Event(RetrieveNewWorkflows, stateData) =>
+    case Event(RetrieveNewWorkflows, _) =>
```

Comment:
TOL do we want to encode "in the middle of checking what workflows to start" into the FSM state/data to avoid multiple `RetrieveNewWorkflows`es being started at once?

Reply:
It seemed to me that the scenario is not possible because we regulate the rate of `RetrieveNewWorkflows` using a `newWorkflowPollRate` timer that only starts counting down after the previous cycle completed.
https://github.com/broadinstitute/cromwell/blob/7229552fd406f7ae4c76e5fd971f0e3c6b10d185/engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala#L338-L340

In addition, there is no special state for the time we are waiting for the workflow store between sending `FetchRunnableWorkflows` and receiving `NoNewWorkflowsToStart`/`NewWorkflowsToStart`.

That said, the above is simply explaining the pattern matching I did against existing code, if the pattern is not right I am happy to go further. I am looking to get more knowledgeable about these state machines.

Reply:
Seems ok to me. Maybe this doesn't really need to be an FSM after all, but this is definitely not a request to go back and change THAT.